tests/trees/test_hoeffding_adaptive_tree_regressor.py

```diff
@@ -78,27 +78,28 @@ def test_hoeffding_tree_perceptron():
         learner.partial_fit(X, y)
         cnt += 1
 
-    expected_predictions = array('d', [525.7553636732247, 352.8160300365902, 224.80744320456478,
-                                       193.72837054292074, 132.6059603765031, 117.06974933197759,
-                                       114.53342429855932, 89.37195405567235, 57.85335051891305,
-                                       60.00883955911155, 47.263185779784266, 25.17616431074491,
-                                       17.43259526890146, 47.33468996498019, 22.83975208548138,
-                                       -7.659282840823236, 8.564101665071064, 14.61585289361161,
-                                       11.560941733770441, 13.70120291865976, 1.1938438210799651,
-                                       19.01970713481836, 21.23459424444584, -5.667473522309328,
-                                       -5.203149619381393, 28.726275200889173, 41.03406433337882,
-                                       27.950322712127267, 21.267116786963925, 5.53344652490152,
-                                       6.753264259267268, -2.3288137435962213, -10.492766334689875,
-                                       -11.19641058176631, -20.134685945295644, -19.36581990084085,
-                                       -38.26894947177957, -34.90246284430353, -11.019543212232008,
-                                       -22.016714766708127, -18.710456277443544, -20.5568019328217,
-                                       -2.636583876625667, 24.787714491718187, 29.325261678088406,
-                                       45.31267371823666, -48.271054430207776, -59.7649172085901,
-                                       48.22724814037523])
+    expected_predictions = array('d', [-106.84237763060068, -10.965517384802226,
```

Comment:
Please, refer to my first point discussed in the main discussion section for a detailed response. The performance changed because previously multiple learning nodes were accessing the same weight matrices (from their ancestors), rather than actually performing a copy of the inherited weights at the moment of their own initialization.

Reply:
Thanks for the clarification 👍. Just wanted to pinpoint the root cause for this difference, given that there are multiple elements in the PR, some not even related to HATR.